### ISSUE TITLE
forwarded accessibility question/complaint

### ISSUE THREAD

**Issue by brianharvey** [jrincayc/ucblogo-code#181]
Anyone want to answer this? :(

From: Jackie McBride <abletec@gmail.com>
Date: Fri, 15 Mar 2024 23:26:36 -0700
Message-ID: <CAM+Q2c5nxCpWH2rNc1ix1LGXh29iUY9tegN8ZTNKuOuiMsa5qA@mail.gmail.com>
Subject: accessibility (or rather lack thereof) of UCB Logo for Windows
To: bh@cs.berkeley.edu

Does Berkley Logo *really* have to be inaccessible? Have you ever given consideration to the fact that something even as simple as Turtle Graphics can actually help blind folks, including children, make art, diagrams, charts, maps, etc? Not to mention learning concepts of directions for mobility training & route memorization? Plus it can assist sight-impaired kids in learning geometric concepts. As it is now, we can't use the editor that comes w/it, nor read the screen output, apparently.

I'm not asking for special facilities like in P5.js editor (though those would be nice), just to be able to use the IDE.

Jackie McBride
Author 36: Last Hours of a Life
Be a hero. Fight Scams. Learn how at [www.scam911.org/](http://www.scam911.org/)
Also check out [brightstarsweb.com](http://brightstarsweb.com/) & [mysitesbeenhacked.com](http://mysitesbeenhacked.com/)

**Comment by dmalec** [comment]
I've read this a few times, mulled it over, and I have some thoughts.

One thing I am hearing loud and clear is that UCB Logo is not accessible for blind and visually impaired folks. I think accessibility in general is important and software accessibility is too often in rough shape even though we have some amazing tools to make it happen in 2024. I would not claim to *be* an advocate for accessibility in software; but, I *have* advocated for software accessibility in my day job and have some empathy for how exhausting that can be. Too often, it can feel like one is making the same points repeatedly in a world that just isn't listening.

One of the points I think folks outside the inner workings of the source code of UCB Logo can undervalue is how a feature which sounds simple is actually rather complicated to implement. I am concerned that making the terminal window compatible with screen readers is one of those things. I'm not suggesting that's all that needs attention; but, I think it is the first step given the feedback. I'm also not suggesting it's impossible, just that I think it needs careful thought and attention to do.

Normally, I would focus on just those two points and not worry about tone, since I think tone policing can silence/discourage folks who need to be heard more. The challenge I see here is that I think it's *also* important to speak up for open source software maintainers. I think we're legitimately seeing a rate of burnout which is worrying and I think speaking up is step one in stopping that.

I am concerned when people look at what is done or isn't done in UCB Logo and feel comfortable that they can assess the values of the people who built or are maintaining UCB Logo. I think it's important for folks to understand that UCB Logo, and so many other open source projects, are built and maintained by a relatively small number of people who do this without compensation. I feel badly that I couldn't contribute more to UCB Logo last year; but, I absolutely had to prioritize my family given things that happened. I lost friends over this prioritization and, honestly, it doesn't feel great to find out I'm also being judged for not building accessibility into UCB Logo. Obviously, I can't control that; but, it still doesn't feel great.

From my perspective, I think it's important to emphasize that if someone wants a feature and has the time, the project welcomes working code contributions. I understand this isn't a panacea; many people may want to positively influence UCB Logo while not being C/C++ programmers. That does take me right back to my thoughts on there being a relatively small number of people keeping open source software running who we don't want to burnout.

To wrap it all up, I don't have a perfect solution. I think my thoughts might be summarized as:
1. I agree accessibility is important and UCB Logo can improve on this front.
2. I can't commit to when I'll have time to look into this given other commitments.
3. Contributions of working code are welcome.

**Comment by pmetzger** [comment]
I think accessibility is a nice idea. If Jackie McBride would like to contribute some patches that would be a good place to start.

**Comment by jrincayc** [comment]
Hm, patches or some more detailed suggestions as to what is not working or how it might be fixed would be useful. I think accessibility on Windows would be good, but I am not sure what needs to be fixed or how to fix it.

**Comment by pmetzger** [comment]
I think this doesn't need to be kept open. Anyone who wishes to submit patches can do so.

**Comment by jrincayc** [comment]
Okay, if this issue does not get enough detail to replicate (as in UCBLogo version and platform, and the screen reader (or other tool) that was tried), or enough details that someone could create a patch (UCBLogo could fix accessibility by doing ...) I think this can be closed in in May.

**Comment by n1ywb-1** [comment]
i sent them an email inviting them to try the web port. i’m sure it’s not ideal but at least it should work with a screen reader and it’s all responsive and scalable; just bounce off ctrl-+ until you can see it ok.

this is why i didn’t use canvas ;)

**Comment by pmetzger** [comment]
@jrincayc It's well past May, 2024.

**Comment by jrincayc** [comment]
> [@jrincayc](https://github.com/jrincayc) It's well past May, 2024.

You will note, that as a computer programmer with a healthy fear of deadlines, I very carefully did not specify the year ;)

Yes, I don't think this is fixable with the information given, so unfortunately it is time to close it.
